amida-tech/amida-auth-microservice
# Feature/ser 280 require email confirmation

If this PR fixes a bug, you _must_ add test cases representative of the bug.

#### What's this PR do?
* Adds a model for migrating the database (`20190408115030-add-validation-columns-users.js`)
* Adds methods for verifying a user's email address
* Rejects requests to the login and reset password endpoints for users when the env vars for requiring account verification are true.
#### Related JIRA tickets:
- [ORANGE-1211](https://jira.amida.com/browse/ORANGE-1211)
#### How should this be manually tested?
The best way to test this flow to spin up a copy of [Orange-Web PR #85](https://github.com/amida-tech/orange-web/pull/85).
* Test the verification process (You can start this process at `https://_webPortalURL_/verify-email`)
* Check to ensure login and password reset are disabled for unverified accounts (when the env is configured to require it)
## 🚨 NEW ENV VARIABLES:
* `AUTH_SERVICE_REQUIRE_ACCOUNT_VERIFICATION`, required, boolean
* `AUTH_SERVICE_REQUIRE_SECURE_ACCOUNT_VERIFICATION`, required, boolean


#### Any background context you want to provide?
#### Screenshots (if appropriate):
